libsolidity/ast/ASTAnnotations.h

```diff
@@ -128,6 +128,14 @@ struct TypeDeclarationAnnotation: DeclarationAnnotation
 	std::string canonicalName;
 };
 
+struct StructDeclarationAnnotation: TypeDeclarationAnnotation
+{
+	/// Whether the struct is recursive, i.e. if the struct type itself occurs again as base type of a dynamic array or
```

Comment:
So it is false if it is a struct that merely contains a recursive struct among its members?
Can you also talk about direct recursion?

Reply:
Direct recursion immediately causes an error - I can mention that here...

Reply:
It's actually not that easy to phrase this correctly - it'll also be true for mutually recursive structs :-).

Reply:
I tried to elaborate a bit more in the comment now.